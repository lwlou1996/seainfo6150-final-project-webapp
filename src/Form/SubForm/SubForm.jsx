import React from 'react'
import styles from "../SubscribeForm/SubcribeForm.module.css";

const SubForm = () => {

    return(

        <div>

            <form>
                <fieldset>
                    <legend><h2>Subscribe Form</h2></legend>
                    <div>
                        <label htmlFor="first-name">
                            First name<sup>*</sup>:
                            <input
                                className={styles.name}
                                type="text"
                                id="first-name"
                                name="first-name"
                                required
                            />
                        </label>
                    </div>
                    <div>
                        <label htmlFor="last-name">
                            Last name<sup>*</sup>:
                            <input
                                className={styles.name}
                                type="text"
                                id="last-name"
                                name="last-name"
                                required
                            />
                        </label>
                    </div>
                    <div>
                        <label htmlFor="email">
                            Email<sup>*</sup>:
                            <input
                                className={styles.email}
                                type="email"
                                id="email"
                                name="email"
                                placeholder="ex: myname@example.com"
                                required
                            />
                        </label>
                    </div>
                    <div>
                        <label htmlFor="telephone">
                            Telephone:
                            <input
                                className={styles.phone}
                                type="tel"
                                id="telephone"
                                name="telephone"
                                pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
                                placeholder="(xxx) xxx xxxx"
                            />
                        </label>
                    </div>
                    <div>
                        <label htmlFor="period">
                            Period:
                            <select id="period" className={styles.period}>
                                <option>please select one</option>
                                <option value="One Week">A Week</option>
                                <option value="One Month">A Month</option>
                                <option value="Three Month">Three Months</option>
                                <option value="Six Months">Six Months</option>
                                <option value="One Year">A Year</option>
                            </select>
                        </label>

                    </div>
                    <div className={styles.typeContainer}>
                        <label className={styles.typeLabel}>Type:</label>
                        <input className={styles.typeOption} type="radio" value="Normal" name="Normal" /><label>Normal</label>
                        <input className={styles.typeOption} type="radio" value="Premium" name="Premium" /><label>Premium</label>
                        <input className={styles.typeOption} type="radio" value="Free Trial" name="Free Trial" /><label>Free Trial</label>
                    </div>
                    <div className={styles.formButtonContainer}>
                        <input className={styles.registerButton} type="submit" value="Register" />
                        <button className={styles.cancelButton}>Cancel</button>
                    </div>
                </fieldset>
            </form>
        </div>

    );

}

export default SubForm;