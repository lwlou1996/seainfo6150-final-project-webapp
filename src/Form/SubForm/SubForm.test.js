import React from 'react'
import {render} from "@testing-library/react";
import SubForm from "./SubForm";

describe('sub form test', () => {

    it('render successfully ', () => {
        const {container} = render(<SubForm />);
        expect(container).toMatchSnapshot();
    });

})