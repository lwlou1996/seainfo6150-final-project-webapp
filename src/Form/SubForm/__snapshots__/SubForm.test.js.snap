// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sub form test render successfully  1`] = `
<div>
  <div>
    <form>
      <fieldset>
        <legend>
          <h2>
            Subscribe Form
          </h2>
        </legend>
        <div>
          <label
            for="first-name"
          >
            First name
            <sup>
              *
            </sup>
            :
            <input
              class="name"
              id="first-name"
              name="first-name"
              required=""
              type="text"
            />
          </label>
        </div>
        <div>
          <label
            for="last-name"
          >
            Last name
            <sup>
              *
            </sup>
            :
            <input
              class="name"
              id="last-name"
              name="last-name"
              required=""
              type="text"
            />
          </label>
        </div>
        <div>
          <label
            for="email"
          >
            Email
            <sup>
              *
            </sup>
            :
            <input
              class="email"
              id="email"
              name="email"
              placeholder="ex: myname@example.com"
              required=""
              type="email"
            />
          </label>
        </div>
        <div>
          <label
            for="telephone"
          >
            Telephone:
            <input
              class="phone"
              id="telephone"
              name="telephone"
              pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
              placeholder="(xxx) xxx xxxx"
              type="tel"
            />
          </label>
        </div>
        <div>
          <label
            for="period"
          >
            Period:
            <select
              class="period"
              id="period"
            >
              <option>
                please select one
              </option>
              <option
                value="One Week"
              >
                A Week
              </option>
              <option
                value="One Month"
              >
                A Month
              </option>
              <option
                value="Three Month"
              >
                Three Months
              </option>
              <option
                value="Six Months"
              >
                Six Months
              </option>
              <option
                value="One Year"
              >
                A Year
              </option>
            </select>
          </label>
        </div>
        <div
          class="typeContainer"
        >
          <label
            class="typeLabel"
          >
            Type:
          </label>
          <input
            class="typeOption"
            name="Normal"
            type="radio"
            value="Normal"
          />
          <label>
            Normal
          </label>
          <input
            class="typeOption"
            name="Premium"
            type="radio"
            value="Premium"
          />
          <label>
            Premium
          </label>
          <input
            class="typeOption"
            name="Free Trial"
            type="radio"
            value="Free Trial"
          />
          <label>
            Free Trial
          </label>
        </div>
        <div
          class="formButtonContainer"
        >
          <input
            class="registerButton"
            type="submit"
            value="Register"
          />
          <button
            class="cancelButton"
          >
            Cancel
          </button>
        </div>
      </fieldset>
    </form>
  </div>
</div>
`;
